include/kernel.h

```diff
@@ -2560,6 +2557,46 @@ static inline void k_work_submit_to_queue(struct k_work_q *work_q,
 	}
 }
 
+/**
+ * @brief Submit a work item to a user mode workqueue
+ *
+ * Sumbits a work item to a workqueue that runs in user mode. A temporary
+ * memory allocation is made from the caller's resource pool which is freed
+ * once the worker thread consumes the k_work item. The workqueue
+ * thread must have memory access to the k_work item being submitted. The caller
```

Comment:
Could we validate this here?
E.g. the item shall bot be submitted if the work-queue thread has no access to it.

Reply:
I don't see a way to robustly do this. Even if we made a check, the configuration could change in between when the item was submitted and when the thread actually processes it.

Reply:
Probably an amendment to the docstring for k_object_access_revoke()